redefinition of typedef ‘git_worktree’ error on 0.26.4

This is an error on gcc 4.3. In the past, libgit2 has fixed a similar issue (libgit2/libgit2/pull/355)

Relevant log:
```
gcc -pthread -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/home/ouruser/local/include -fPIC -I/usr/local/include -I/home/ouruser/check-git/venv_check_git/include -I/home/ouruser/local/include/python3.5m -c src/repository.c -o build/temp.linux-x86_64-3.5/src/repository.o
    In file included from src/repository.c:41:
    src/worktree.h:43: error: redefinition of typedef ‘git_worktree’
    /home/ouruser/local/include/git2/types.h:108: error: previous declaration of ‘git_worktree’ was here
    error: command 'gcc' failed with exit status 1
```

It's unfortunate that the minimum compiler version changed between 0.26.3 and 0.26.4. Is this a bug?

--- a/src/worktree.h
+++ b/src/worktree.h
@@ -33,14 +33,14 @@
 #include <git2.h>
 #include <git2/worktree.h>
 
-typedef struct git_worktree {
+struct git_worktree {
 	char *name;
 	char *gitlink_path;
 	char *gitdir_path;
 	char *commondir_path;
 	char *parent_path;
 	int locked:1;
-} git_worktree;
+};
 
 PyObject* wrap_worktree(Repository* repo, git_worktree* wt);
 

